x/pool-incentives/keeper/grpc_query.go

```diff
@@ -157,6 +158,45 @@ func (q Querier) IncentivizedPools(ctx context.Context, _ *types.QueryIncentiviz
 		}
 	}
 
+	// Retrieve the migration records between balancer pools and concentrated liquidity pools.
+	// This comes from the superfluid keeper, since superfluid is the only pool incentives connected
+	// module that has access to the gamm modules store.
+	migrationRecords, err := q.superfluidKeeper.GetAllMigrationInfo(sdkCtx)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	// Iterate over all migration records.
+	for _, record := range migrationRecords.BalancerToConcentratedPoolLinks {
+		linkedClPoolIsIncentivized := false
+		var duration time.Duration
+
+		// If the linked concentrated liquidity pool is in the list of incentivized pools,
+		// note this and add its balancer counterpart to the list of incentivized pools.
+		for _, incentivizedPool := range incentivizedPools {
+			if incentivizedPool.PoolId == record.ClPoolId {
+				linkedClPoolIsIncentivized = true
+				duration = incentivizedPool.LockableDuration
+				continue
+			}
+		}
```

Comment:
Consider creating a `map[uint64]struct{}` of incentivized pool IDs in the first loop above. This would allow checking the map here in O(1) instead of repeatedly iterating all incentivized pools for every link

Reply:
Good call, added here https://github.com/osmosis-labs/osmosis/pull/5589/commits/bba673dbd0e0489915fa88aba713329c4fde8895